flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -581,16 +608,26 @@ static RowType toFlinkRowType(Schema schema, TableSchema requestedSchema) {
 
   static IcebergStreamWriter<RowData> createStreamWriter(
       Table table,
+      TableLoader tableLoader,
       FlinkWriteConf flinkWriteConf,
       RowType flinkRowType,
-      List<Integer> equalityFieldIds) {
+      List<Integer> equalityFieldIds,
+      long minReloadIntervalMs) {
```

Comment:
My comment might be lost somewhere, but I think it would be better to receive the TableSupplier here, instead of creating a new "incidentally" same supplier than we do in the FlinkSink.

Also seeing the TableLoader/TableSupplier in one place mad me think how do we separate the functionality of the 2. Do we really need 2 objects providing/reloading tables?